src/foam/nanos/auth/AbstractAuthService.js

```diff
@@ -0,0 +1,21 @@
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'AbstractAuthService',
+  abstract: true,
+
+  implements: [
+    'foam.nanos.auth.AuthService'
+  ],
+  
+  methods: [
+    {
+      name: 'require',
+      javaCode: `
+        if ( ! check(x, permission) ) {
+          if ( exception != null ) throw exception;
+          throw new AuthorizationException();
```

Comment:
Why not add the permission that was checked into the exception? It doesn't have to be in the message, but if we know the failing permission it may help debug things in the future.

Reply:
I think this is a good idea. I don't think it should be sent to the client, but if the property is transient then it doesn't matter.

The same thing could be accomplished with an AuthService decorator at the end of the chain that adds the permission too - what do you think of this approach?

Reply:
Sounds good to me .. @jlhughes added some more comments on this, so make sure to sync up on those, which sounds like it will incorporate this into it as well.